webrender/src/platform/macos/font.rs

```diff
@@ -497,13 +505,22 @@ impl FontContext {
         // At the end of all this, WR expects individual RGB channels and ignores alpha
         // for subpixel AA.
         // For alpha/mono, WR ignores all channels other than alpha.
-        // Also note that WR expects text to be black bg with white text, so invert
-        // when we draw the glyphs.
-        let (antialias, smooth, bg_color) = match font.render_mode {
-            FontRenderMode::Subpixel => (true, true, 1.0),
-            FontRenderMode::Alpha => (true, false, 1.0),
-            FontRenderMode::Bitmap => (true, false, 0.0),
-            FontRenderMode::Mono => (false, false, 1.0),
+        // Also note that WR expects text to be white text on black bg, so invert
+        // when we draw the glyphs as black on white.
+        let use_white_on_black = self.should_use_white_on_black(font.color);
+        let (antialias, smooth, text_color, bg_color, bg_alpha, invert) = match font.render_mode {
```

Comment:
This could possibly be simplified by deriving some of those values after the match, but it can be done as a follow up, if at all.